Train Barrier Feature (Finished)

ExtremeMakerX opened the pull request:
### This adds the option of train Barriers for the R179, Changes the spacing from int to float and Widget Checkbox on the main siding screen to enable or disable train Barriers.
The HAS_TRAIN_BARRIER Custom Resource command is not usable

![2022-02-08_22 11 41](https://user-images.githubusercontent.com/79700465/153118533-2c3d7249-a089-4f85-bff8-6bfce9f1d516.png)
![2022-02-08_22 12 25](https://user-images.githubusercontent.com/79700465/153118536-e71b343d-7396-4905-a415-a26c26706bfc.png)
![2022-02-08_22 12 36](https://user-images.githubusercontent.com/79700465/153118537-96c3c517-163d-4d3c-b7b3-3b68fb4bcab3.png)

jonafanho commented:
@ExtremeMakerX can you please explain a little bit what your code does? I would like to make this code more generic so they can be applied to the other trains as well.

ExtremeMakerX commented:
> @ExtremeMakerX can you please explain a little bit what your code does? I would like to make this code more generic so they can be applied to the other trains as well.

Ok hold on. I will tell you soon

ExtremeMakerX commented:
> @ExtremeMakerX can you please explain a little bit what your code does? I would like to make this code more generic so they can be applied to the other trains as well.

Okay feel free to let me know if I didn't explain enough.

So what my code is the R179_TB in TrainClientRegistry is invisible in the train selection is because in KEY_ORDERS I added in the !trainBarrierOption which basically allows any train with the trainBarrierOption enabled will not show in the train selection. I added trainBarrierlength which of course allows float values on train spacing. In the siding menu when The TrainBarrierWidget is checked. The siding gets updated only if there is a R179 on the siding track then it replace the R179 with the R179_TB and then when unchecked goes back to the R179. Train barrier that is on 2 sides is rendered like a gangway connection but without players going to other cars. Reason the sides, up and down are transparent is because there is transparent connection textures. Also reason the train barrier is in the siding script is so the TB widget if checked can be saved when you open up the siding again. Hopefully I helped you understand what's going on.
